lib/samlr/tools.rb

```diff
@@ -135,10 +136,13 @@ def self.parse(data, compressed: false)
 
     def self.inflate(data)
       inflater  = Zlib::Inflate.new(-Zlib::MAX_WBITS)
-      decoded = inflater.inflate(data)
-
-      inflater.finish
-      inflater.close
+      begin
+        decoded = inflater.inflate(data)
+        inflater.finish
+        inflater.close
+      rescue
```

Comment:
can we rescue specific exceptions here? It's not a great idea to catch everything.